### PR TITLE
fix(capture): allow empty scoped target creation

### PR DESCRIPTION
**Pull request by chhoumann**
Allow Capture target pickers to create a new note even when the scoped file list starts empty.

The current runtime had the right shared primitive already: folder, tag/filter, and property capture targets all route through `InputSuggester` with a create row when **Create file if it doesn't exist** is enabled. The bug was that both capture picker helpers aborted before reaching that shared picker if the scoped list was empty. This moves the empty-list guard behind the create-enabled check, keeps create-disabled scopes as clear errors, and marks empty create-enabled one-page target requirements as runtime-only so the runtime picker keeps duplicate suppression and path normalization.

I also hardened `InputSuggester` tab completion for empty suggestion lists and updated the Capture docs to remove the old limitation that a matching Markdown file had to exist before the picker could open.

Verification evidence:
- Reproduced before the fix in the isolated vault: empty folder scope showed `Folder __qa_empty_scope/ is empty.` and created no file; empty tag scope showed `No files matched the capture target filters.` with no picker.
- Live after-fix folder proof: empty scoped folder opened a picker with `Choose a note or type to create one`, showed `Create new note: First Created`, created `__qa_live_empty_1782052468479/First Created.md`, and wrote `empty proof`.
- Live non-regression proof: non-empty folder picker still selected `Existing.md`, updated it to include `existing proof`, and did not create a duplicate.
- Live tag proof: empty tag scope showed the create row and created `Tag Created 1782052544480.md` with `tag proof`.
- Live property proof: empty property scope showed the create row and created `Property Created 1782052566480.md` with `property proof`.
- `pnpm exec tsc -noEmit -skipLibCheck`
- `pnpm run lint`
- `pnpm run check` (0 errors, existing warnings only)
- `pnpm test`
- `pnpm run build`
- `pnpm run obsidian:e2e -- dev:errors` returned no captured errors.

Release impact: no migration. Existing create-disabled empty scopes still abort instead of opening an unusable picker.

---

<a href="https://app.devin.ai/review/chhoumann/quickadd/pull/1402" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open in Devin Review
  </picture>
</a>

## Summary by CodeRabbit

* **New Features**
  * File picker now opens for empty destination scopes when **Create file if it doesn’t exist** is enabled, so you can type the first note name to create.
  * Capture-target selection now supports creating a target even when no existing matches are found.
* **Bug Fixes**
  * Improved keyboard/tab behavior in suggestion inputs to avoid crashes when no suggestions are available.
* **Documentation**
  * Updated capture-target documentation and clarified non-interactive CLI handling of `missingFlags`.
* **Tests**
  * Added coverage for empty-scope capture-target resolution and keyboard/suggestion edge cases.

### CODE REVIEW

File: src/engine/CaptureChoiceEngine.ts
[the remark this answers is not in the document]
Comment (chhoumann) [reply src/engine/CaptureChoiceEngine.ts:1]: Verified and addressed in b620325a. `#tag` capture targets are handled by `parseCaptureFileFilterTarget` and the filter path, so I removed the unreachable `tag` union branch plus the stale `selectFileWithTag` helper/import.

File: src/preflight/collectChoiceRequirements.ts [hunk 1]
```diff
@@ -330,16 +330,31 @@ async function collectForCaptureChoice(
...
+			collector.requirements.set(QA_INTERNAL_CAPTURE_TARGET_FILE_PATH, {
+				id: QA_INTERNAL_CAPTURE_TARGET_FILE_PATH,
+				label: "Select capture target file",
+				type: "file-picker",
+				source: "collected",
+				options,
+				displayOptions,
+				runtimeOnly: true,
+				placeholder: "Type a new note name in the capture target picker",
+			});
```
[the remark this answers is not in the document]
Comment (chhoumann) [reply src/preflight/collectChoiceRequirements.ts:408]: Verified before changing this. Non-interactive CLI runs should continue to report unresolved runtime-only inputs instead of opening prompts, but the conclusion that there is no flag path was not correct: the returned `missingFlags` entry is `value-__qa.captureTargetFilePath=<value>`, which seeds `QA_INTERNAL_CAPTURE_TARGET_FILE_PATH` and is consumed by the capture engine before opening the picker. I added regression coverage for that path and documented that generated `missingFlags` entries should be passed back exactly.
